net/api/impl/apple_client.mm

```diff
@@ -57,21 +57,23 @@ of this software and associated documentation files (the "Software"), to deal
     if (!user_agent.empty())
       [request setValue:[NSString stringWithUTF8String:user_agent.c_str()] forHTTPHeaderField:@"User-Agent"];
 
-    if (!post_body.empty()) {
-      request.HTTPBody = [NSData dataWithBytes:post_body.data() length:post_body.size()];
+    if (is_post) {
       request.HTTPMethod = @"POST";
-    } else if (!post_file.empty()) {
-      NSError * err = nil;
-      NSString * path = [NSString stringWithUTF8String:post_file.c_str()];
-      const unsigned long long file_size = [[NSFileManager defaultManager] attributesOfItemAtPath:path error:&err].fileSize;
-      if (err) {
-        http_response_code = -1;
-        NSLog(@"Error %d %@", static_cast<int>(err.code), err.localizedDescription);
-        return false;
+      if (!post_file.empty()) {
+        NSError * err = nil;
+        NSString * path = [NSString stringWithUTF8String:post_file.c_str()];
+        const unsigned long long file_size =
+          [[NSFileManager defaultManager] attributesOfItemAtPath:path error:&err].fileSize;
+        if (err) {
+          http_response_code = -1;
+          NSLog(@"Error %d %@", static_cast<int>(err.code), err.localizedDescription);
+          return false;
+        }
+        request.HTTPBodyStream = [NSInputStream inputStreamWithFileAtPath:path];
+        [request setValue:[NSString stringWithFormat:@"%llu", file_size] forHTTPHeaderField:@"Content-Length"];
+      } else {
+        request.HTTPBody = [NSData dataWithBytes:post_body.data() length:post_body.size()];
```

Comment:
It will POST a request with empty body, Iwould assume. Any reason it would
not?

On Monday, March 23, 2015, Alex Zolotarev <notifications@github.com> wrote:

> In net/api/impl/apple_client.mm
> <https://github.com/KnowSheet/Bricks/pull/138#discussion_r26956250>:
>
> > -        NSLog(@"Error %d %@", static_cast<int>(err.code), err.localizedDescription);
> > -        return false;
> > +      if (!post_file.empty()) {
> > +        NSError * err = nil;
> > +        NSString * path = [NSString stringWithUTF8String:post_file.c_str()];
> > +        const unsigned long long file_size =
> > +          [[NSFileManager defaultManager] attributesOfItemAtPath:path error:&err].fileSize;
> > +        if (err) {
> > +          http_response_code = -1;
> > +          NSLog(@"Error %d %@", static_cast<int>(err.code), err.localizedDescription);
> > +          return false;
> > +        }
> > +        request.HTTPBodyStream = [NSInputStream inputStreamWithFileAtPath:path];
> > +        [request setValue:[NSString stringWithFormat:@"%llu", file_size] forHTTPHeaderField:@"Content-Length"];
> > +      } else {
> > +        request.HTTPBody = [NSData dataWithBytes:post_body.data() length:post_body.size()];
>
> What if post_body will be empty here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/Bricks/pull/138/files#r26956250>.
>


-- 
Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783